### PR TITLE
Fix integration tests failing due to non-empty test folders

### PR DESCRIPTION
**Pull request by quantumagi**
This PR:
- Separates existing code into the "CleanupTestFolder" method.
- Ensures the method is used where required.
- Fixes CanListIndexesFromRPC to ignore order of results.

### CODE REVIEW

File: src/Stratis.Bitcoin.IntegrationTests/NodeBuilder.cs [hunk 1]
```diff
@@ -328,55 +328,50 @@ public override string ToString()
...
+                catch (DirectoryNotFoundException)
+                {
+                    return true;
+                }
+                catch (Exception)
+                {
+                }
+                if (!tryKill)
+                    return false;
+                tryKill = false;
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/NodeBuilder.cs]: Is the purpose of tryKill to try deleting one more time?
If that's the case, then having a while(true) + a flag (tryKill) seems a bit overengineered if all we want to do is try to delete something twice. In this case, a for loop with a retryCount of 2 seems good enough.

File: src/Stratis.Bitcoin.IntegrationTests/NodeBuilder.cs [hunk 2]
```diff
@@ -328,55 +328,50 @@ public override string ToString()
...
-                //}
                 Thread.Sleep(1000);
-                Directory.Delete(caller, true);
             }
+        }
+
+        public static NodeBuilder Create([CallerMemberNameAttribute] string caller = null, string version = "0.13.1")
+        {
+            version = version ?? "0.13.1";
+            if (!Directory.Exists("TestData"))
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/NodeBuilder.cs]: no need to check `if (!Directory.Exists("TestData"))` as `Directory.CreateDirectory` will create the directory only if they exist.

File: src/Stratis.Bitcoin.IntegrationTests/NodeBuilder.cs [hunk 3]
```diff
@@ -328,55 +328,50 @@ public override string ToString()
...
-                //	}
-                //}
                 Thread.Sleep(1000);
-                Directory.Delete(caller, true);
             }
+        }
+
+        public static NodeBuilder Create([CallerMemberNameAttribute] string caller = null, string version = "0.13.1")
+        {
+            version = version ?? "0.13.1";
```
[the remark this answers is not in the document]
Comment (quantumagi) [reply src/Stratis.Bitcoin.IntegrationTests/NodeBuilder.cs]: The github comparison is a bit wonky as I did not change or add that code. Fixed it however. :)

File: src/Stratis.Bitcoin.IntegrationTests/NodeBuilder.cs (hunk 2 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.IntegrationTests/NodeBuilder.cs]: Same here.

**Pull request merged**